astropy/units/format/vounit.py

```diff
@@ -221,6 +221,10 @@ def to_string(
                 "represent scale for dimensionless units. "
                 f"Multiply your data by {unit.scale:e}."
             )
+        if unit.physical_type == "dimensionless":
```

Comment:
This is not correct, as it would also capture other dimensionless units, e.g.,
```
In [2]: u.percent.physical_type
Out[2]: PhysicalType('dimensionless')
```
...
Actually, I realize that is taken care already in the `if` statement above, but let's at least combine the two, i.e.,
```
if unit.physical_type == "dimensionless":
    if unit.scale != 1:
        raise UnitScaleError(...)
    # <your comment>
    return "1"
```